### PR TITLE
Hadoop Cluster

### LINKED ISSUE #44: Hadoop Cluster
**MicroByte Title:**  Hadoop Cluster

**Domain:**  Big data management

**Scope of learning for learners:** 
- Creating Namenode(masternode)
- Creating Datanode(slavenode)
- Creating Clientnode
- Uploading and Reading file in Hadoop
- Reading files in a terminal and in WebUi of Hadoop

**Activities:** 
- Activity 1 ( Creating Namenode ): contains the explanation of how we can create Namenode by configuring the XML files in Hadoop and contains **Task** for the learners.
- Activity 2 ( Starting Namenode ): contains the explanation of how we can Start Datanode and check whether the datanode is started or not. Contains **Task** for the learners.
- Activity 3 ( Creating Datanode ): contains the explanation of how we can Create Datanode and it contains **Task** for the learners.
- Activity 4 ( Starting Datanode ): contains an explanation of how we can start Datanode and it contains **Task** for the learners.
- Activity 5:  checked datanodes are connected with master and given the task to connect two or more Datanodes.
- Activity 6 : ( creating Clientnode) : contains an explanation of how we can create Client node and contains **Task** for the learners.
- Activity 7: Uploaded and read files in Hadoop.
- Activity 8 (Checking file in Hadoop) : contains the explanation of how we can check files that we have uploaded in Hadoop by the terminal as well as in WebUI.

### PR DESCRIPTION
**Pull request by Nitesh-thapliyal**
### Checklist: 

- [x] Added activities for learners
- [x] Didn't give the solutions to the added activities straight away
- [x] Followed a LearnByDoing approach
- [x] Clean and well documented code (if source code files are involved)
- [x] Added images for better understanding if deemed necessary

### CODE REVIEW

**Review by YourAKShaw — CHANGES_REQUESTED** [review]
@Nitesh-thapliyal You content LGTM! But, I have a few requests:

- Please do a proof reading of your `README.md` as it's kinda has quite a few typos.
- Kindly add some applications of this MicroByte, as to what will be the learner's benefit once they're done with it. This will add to the further incentive of their learning.
- Also, the applications **should be relevant** with respect to the given activities.

_Needless to say, you've done a very fine job till now, just a few knick-knacks here and there._

**Review by YourAKShaw — APPROVED** [review]
@Nitesh-thapliyal Yet again you've done a pretty fine job. Let's see what the others have to say. 

Further more, once this potential PR merges, kindly try to create a few issues on which a creator or learner can work on to make this a seamless content. Always remember, improvements are always possible.

File: Hadoop Cluster/README.md
Comment (amoghrajesh) [inline Hadoop Cluster/README.md]: This looks like this is the command, correct it to: `mkdir /directory-name`

**Review by amoghrajesh — APPROVED** [review]
Hey, @Nitesh-thapliyal I liked the content.
Amazing content put up in an engaging way.

- You just have some small typos here and there and I have added a comment for changing the command which was mistyped i guess.
- Also, you can refer to the Linux and AWS bytes on Crio site for learners to pick up if they are new to these concepts.

Great work! Make these changes, shall approve after :)

**Review by YourAKShaw — CHANGES_REQUESTED** [review]
@Nitesh-thapliyal Sorry that I overlooked these. Kindly make the above mentioned changes and we shall merge it.

**Review by YourAKShaw — APPROVED** [review]

**Comment by sudhanshutiwari264** [comment]
@all-contributors kindly add @Nitesh-thapliyal for:

- [x] code
- [x] content
- [x] ideas
- [x] doc

**Comment by YourAKShaw** [comment]
@Nitesh-thapliyal Kindly showcase your new MicroByte on our slack workspace using [this](https://github.com/Crio-Bytes/Demo-Repo/blob/master/Maintainer%20Resources/new-microbyte-showcase-template.md) template. Also if you've got ideas that'll further improve this MicroByte, you are free to open up some issues on which you or the community can contribute.

**Pull request merged**
